test/source/put.test.js

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-env mocha */
+/* eslint-disable no-param-reassign */
+import assert from 'assert';
+import { promisify } from 'util';
+import zlib from 'zlib';
+import { putSource } from '../../src/source/put.js';
+import { Nock } from '../utils.js';
+
+const gunzip = promisify(zlib.gunzip);
+
+describe('Source PUT Tests', () => {
+  /** @type {import('../utils.js').NockEnv} */
+  let nock;
+  let context;
+
+  beforeEach(() => {
+    context = {
+      attributes: {},
+      env: { HELIX_STORAGE_DISABLE_R2: 'true' },
+      log: console,
+    };
```

Comment:
yep, like said in `get.test.js` above, it's better to create a fresh `context` on every test.